pkg/codegen/nodejs/gen_program_expressions.go

```diff
@@ -579,6 +579,10 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 		g.genCan(w, expr)
 	case "rootDirectory":
 		g.genRootDirectory(w)
+	case "pulumiResourceName":
+		g.Fgenf(w, "pulumi.resourceName(%s)", expr.Args[0])
```

Comment:
Probably a dumb question -- should this be `%s`, or `%v`?

Reply:
Turns out, it could be anything... we're inconsistent about using 's' or 'v' but it turns out the formatting code only checks to see if this is 'o' to set `rhs` mode and any other letter writes the expression string in `lhs` mode.

I'll make it 'v' just to be a bit more consistent.